src/main/kotlin/org/wycliffeassociates/resourcecontainer/ResourceContainer.kt

```diff
@@ -64,20 +62,39 @@ class ResourceContainer private constructor(val file: File, var config: Config?
                 //writeTableOfContents(p)
             }
         }
+        media?.let {
+            writeMedia()
+        }
     }
 
     fun writeManifest() {
         accessor.initWrite()
         accessor.write(MANIFEST_FILENAME) { writeManifest(it) }
     }
 
-    private fun writeManifest(writer: Writer) {
-        val mapper = ObjectMapper(YAMLFactory())
+    private fun writeManifest(writer: OutputStream) {
+        val factory = YAMLFactory()
+        factory.disable(JsonGenerator.Feature.AUTO_CLOSE_TARGET)
+        val mapper = ObjectMapper(factory)
         mapper.registerModule(KotlinModule())
         mapper.setSerializationInclusion(Include.NON_NULL)
-        writer.use {
-            mapper.writeValue(it, manifest)
-        }
+        mapper.writeValue(writer, manifest)
+        writer.flush()
```

Comment:
Not needed to close the stream?

Reply:
Correct, this is exactly the bug fix. The output stream is still being used in a loop, and closed elsewhere. Closing it here will crash and prevent the writing of other zip files in the loop.